clang/lib/CIR/CodeGen/CIRGenTypes.cpp

```diff
@@ -138,7 +139,7 @@ isSafeToConvert(const RecordDecl *rd, CIRGenTypes &cgt,
   if (!alreadyChecked.insert(rd).second)
     return true;
 
-  const Type *key = cgt.getASTContext().getTagDeclType(rd).getTypePtr();
+  const Type *key = cgt.getASTContext().getCanonicalTagType(rd).getTypePtr();
```

Comment:
FYI slightly unrelated, but it might make sense to `assert(rd->isCompleteDefinition()` since some of the uses here only make sense when applied on the definition, such as the `fields` iteration down below.